src/Mvc/Mvc.Core/src/Diagnostics/MvcDiagnostics.cs

```diff
@@ -525,7 +525,7 @@ public AfterControllerActionMethodEventData(ActionContext actionContext, IReadOn
         protected override KeyValuePair<string, object> this[int index] => index switch
         {
             0 => new KeyValuePair<string, object>(nameof(ActionContext), ActionContext),
-            1 => new KeyValuePair<string, object>(nameof(Controller), Controller),
+            1 => new KeyValuePair<string, object>(nameof(Arguments), Arguments),
```

Comment:
@davidfowl I'm not really familiar with how Diagnostics works. Would this break consumers that indexed in to this?

Reply:
I think that the behavior here is wrong, we should fix it, and make an announcement if necessary.

Reply:
It will break consumers and before we merge it, we should find the big consumers and see how hard it will break

Reply:
> It will break consumers and before we merge it, we should find the big consumers and see how hard it will break

How do you propose we do this? Who are the consumers of this data, libraries like App insights and folks like that? 

From what I understand this is diagnostic source, which is very specialized.

Reply:
Application insights, open telemetry, other APM vendors that are consuming this data.

Reply:
Good to know.

Given that's the case, maybe we should probably hold-off on this for the time being since we haven't received an actual customer report where this was problematic.

Reply:
We shouldn't keep a PR around if we don't plan to accept that now. If this is ready to go and it does add value (which based on the issue it's referencing seems to be the case) we should take this. Otherwise - let's close the PR.

Reply:
Fair enough, let's close then